service/service.go

```diff
@@ -0,0 +1,53 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	goaclient "github.com/goadesign/goa/client"
+
+	"github.com/fabric8-services/fabric8-common/errors"
+	"github.com/fabric8-services/fabric8-common/goasupport"
+
+	clusterclient "github.com/fabric8-services/fabric8-cluster-client/cluster"
+)
+
+type Service interface {
+	ClustersUser(ctx context.Context) (*clusterclient.ClusterList, error)
```

Comment:
Can you rename this function to something a little more explicit ? eg: `List` or `ListClusters` ?

Reply:
@xcoulon I kept same name of original client API so it's straight forward mapping and I guess we should keep it consistent.  It's just a wrapper of target client API. https://github.com/fabric8-services/fabric8-cluster-client/blob/master/cluster/user.go#L28

Reply:
ok, sounds reasonable, although, I would prefer some more meaningful method names since you're introducing this wrapper.

Reply:
Regarding the names. The client function `ClustersUser()` is generated by goa from the endpoint path: `/api/user/clusters`. This is why such functions can have ugly names. The service we are introducing here is more user friendly API, so, let's give it proper names :)
I think `List()` or `ListClusters()` would be a good name for such function.

But we actually should not use `/api/user/clusters` at all. That endpoint is supposed to be deprecated and we need to keep it just not to break the current clients of the cluster/auth service. All new functionality should load particular cluster by URL instead. I also just realized that we actually have not implemented that new endpoint yet - https://github.com/fabric8-services/fabric8-cluster/issues/59
It should be our highest priority in the cluster service right now since we really need to start using that endpoint in different places.
Let's create that endpoint in the cluster service and then update this PR to replace `ClustersUser()` by `ClusterByURL()` in this service.

Reply:
I will go with `UserClusters`, we need to use `user` to clarify it's user_specific and not all_clusters.  And `s` is enough rather `list` to indicates multiple.

Reply:
I actually confused /user/clusters and /clusters endpoints here. We are going to keep using /user/clusters endpoint. It's /clusters endpont which we should deprecate and it's not used here.